bblocks/import_tools/weo.py

```diff
@@ -0,0 +1,287 @@
+        Parser(folder).parse_data().to_csv(self._path, index=False)
+
+    def _download_data(self) -> None:
+        """Get data for the version, raise error is no data for version is found"""
+
+        href = get_smdx_href(self.version)
+        if href is None:
+            raise ValueError(
+                f"No data found for version:" f" {self.version[0]}, {self.version[1]}"
+            )
+
+        # get and parse data and save to disk
+        response = get_response(BASE_URL + href)
+        folder = unzip(io.BytesIO(response.content))
+        Parser(folder).parse_data().to_csv(self._path, index=False)
+
+    def load_data(self, indicators: str | list = "all") -> ImportData:
+        """Load data to object
+
+        When called this method will load WEO indicators to the object.
+        If the data is not downloaded, it will download the data. If the data is
+        already downloaded, it will load the data from disk. If the version is set to
+        'latest', it will find the latest version and download the data.
+
+        Args:
+            indicators: The indicators to load. If 'all' is passed, all indicators will be loaded. Default is 'all'
+
+        Returns:
+            same object with data loaded
+        """
+
+        # check if latest is requested
+        if self.version == "latest":
+            self.version = gen_latest_version()  # set latest expected version
+            if not self._path.exists():  # check if not downloaded
+                self._download_latest_data()  # find latest and download data
+
+        else:
+            if not self._path.exists():  # check if not downloaded
+                self._download_data()  # download data
+
+        # check if raw data is not loaded to object
+        if self._raw_data is None:
+            self._raw_data = pd.read_csv(self._path)
+
+        # load indicators
+        if indicators == "all":
+            indicators = self._raw_data["CONCEPT_CODE"].unique()
+        if isinstance(indicators, str):
+            indicators = [indicators]
+
+        self._data.update(
+            {
+                indicator: (
+                    self._raw_data[
+                        self._raw_data["CONCEPT_CODE"] == indicator
+                    ].reset_index(drop=True)
+                )
+                for indicator in indicators
+            }
+        )
+        logger.info("Data loaded successfully")
+        return self
+
+    def update_data(self, reload_data: bool = True) -> ImportData:
+        """Update the data
+
+        When called it will update the data used to load the indicators.
+        If reload_data is True, it will reload the indicators to the object.
+
+        Args:
+            reload_data: If True, reload the indicators after updating the data. Default is True.
+
+        Returns:
+            same object with updated data to allow chaining
+        """
+        if len(self._data) == 0:
+            raise RuntimeError("No indicators loaded")
+
```

Comment:
Since a version (`latest` or an actual version) must be specified when you instantiate the object, there will always be something that you can use in order to figure out what you should update.

I don't think we should require actual indicators to be loaded in order to update the underlying data.... especially since those two things (getting the raw data, and getting the indicators) are separate processes

Reply:
true @jm-rivera . But there is some logic to convert the version from "latest" to an actual version and roll back the version that occurs in `load_data` and elsewhere we raise an error if no indicators have been loaded. In some places this is the case because we query an api eg WDI data. However I think we should have a consistent approach

Reply:
Then that logic should be decoupled from load_data so that it can also be called when the user requests the list of indicators